File: src/main/webapp/resources/frame/data/RoomStateEventHandler.js


export default class RoomStateEventHandler{
    constructor (){
        this._room;
        this._handleEvent;
    }

    spawn(room, handleEvent){
        this._room = room;
        this._handleEvent = handleEvent;
        return this;
    }

    get room(){
        return this._room;
    }

    get handleEvent(){
        return this._handleEvent;
    }
}

File: src/main/java/com/semiprj/data/RoomStateEventHandler.java
package com.semiprj.data;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@ToString
public class RoomStateEventHandler {
    public RoomStateEventHandler(){}
    public RoomStateEventHandler(Room room, HandleEvent handleEvent){
        this.room = room;
        this.handleEvent = handleEvent;
    }
    private Room room;
    private HandleEvent handleEvent;
}
